src/dlgConnectionProfiles.cpp

```diff
@@ -1421,20 +1422,31 @@ void dlgConnectionProfiles::slot_copy_profile()
         return;
     }
 
-    copyFolder(mudlet::getMudletPath(mudlet::profileHomePath, oldname), mudlet::getMudletPath(mudlet::profileHomePath, profile_name));
-    mProfileList << profile_name;
-    slot_item_clicked(pItem);
-    // Clear the Discord optin on the copied profile - just because the source
-    // one may have had it enabled does not mean we can assume the new one would
-    // want it set:
-    discord_optin_checkBox->setChecked(false);
+    QApplication::setOverrideCursor(Qt::BusyCursor);
+    mpCopyProfile->setText(tr("Copying..."));
+    mpCopyProfile->setEnabled(false);
+    auto future = QtConcurrent::run(dlgConnectionProfiles::copyFolder, mudlet::getMudletPath(mudlet::profileHomePath, oldname), mudlet::getMudletPath(mudlet::profileHomePath, profile_name));
```

Comment:
Can the copying fail (e.g. out of file-system space?) If so, how does Mudlet, or rather this background process, recover?

Reply:
Not sure, but it would be the same behaviour as the current code - so out of scope here.

Reply:
:thinking: I'm not sure - if the system breaks whilst it is in the foreground then it is obvious to the user isn't it - whereas if it never completes but is being done "behind the scenes" does it handle reporting back to the user that it has failed?

Reply:
There are no provisions in the current code for this, so I am not going to hold up this improvement further adding them here. It's already been sitting as-is for a year.